src/main/java/core/basesyntax/service/impl/ValidatorImpl.java

```diff
@@ -0,0 +1,42 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.service.Validator;
+import core.basesyntax.service.activity.ActivityHandler;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Predicate;
+import java.util.regex.Pattern;
+
+public class ValidatorImpl implements Validator {
+    private static final int ACTIVITY_TYPE_INDEX = 0;
+    private static final String HEAD = "type,fruit,quantity";
+    private static final String VALID_DATA_PATTERN = "[a-z],[a-z]+,[0-9]+";
+    private final Map<String, ActivityHandler> activityHandlerMap;
+
+    public ValidatorImpl(Map<String, ActivityHandler> activityHandlerMap) {
+        this.activityHandlerMap = activityHandlerMap;
+    }
+
+    @Override
+    public boolean validateData(List<String> inputData) {
+        List<String> dataCopy = new ArrayList<>(inputData);
+        if (inputData.isEmpty() || !checkAndDeleteFirstLine(dataCopy)) {
```

Comment:
what will be in your program according to your boolean statement if inputData is not empty?